### PR TITLE
Build on Boarding screen

### PR DESCRIPTION
**Pull request by KeynesBizimana**
#### What does this PR do?

This pull request introduces significant enhancements and new features to the authentication and security verification processes within the application.

#### Description of Task to be completed?

1. Authentication Screens:
   

- [x]  Added comprehensive authentication screens allowing users to sign in using their email and phone number.

- [x] Implemented seamless registration using Phone number and email 

2. OTP Verification Pages:

- [x]  Developed intuitive OTP verification pages to ensure the security and validity of user-provided phone numbers and email addresses.
- [x]  Strengthened user authentication through multi-factor verification using OTP codes.

3. Fingerprint Scanner Integration Screen:

- [x]  Integrated state-of-the-art fingerprint scanner functionality to elevate the security standards of the application.
- [x]  Users can now opt for fingerprint authentication for a secure and convenient login experience.

Purpose:
These enhancements are designed to significantly enhance user experience, streamline the authentication process, and bolster the security measures of the application. The new features aim to provide users with seamless access to the application.

#### Screenshots

Screenshot_20240410_105012_Expo Go
Screenshot_20240410_105247_Expo Go

### CODE REVIEW

File: src/components/PasswordInput.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+           
+        </View>
+    );
+}
+
+export default PasswordInput;
+
+const styles = StyleSheet.create({
+    input: {
+
```
Comment (Irirwanirira) [inline src/components/PasswordInput.tsx]: Kindly remove the blank spaces present in your code base

File: src/components/PhoneNumberInput.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+   
+    
+    const [countryCode, setCountryCode] = useState('');
+  const [phoneNumber, setPhoneNumber] = useState('');
+
+  const countryCodes = [
+    { label: '+250', value: '+250' },
+      { label: '+400', value: '+400' },
+     { label: '+62', value: '+62' }
+    // Add more country codes as needed
```
Comment (Irirwanirira) [inline src/components/PhoneNumberInput.tsx]: Please remove comments to the code to make your code base clean

File: src/screens/Auth/EmailLoginScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+               
+                
+            </View>
+            
+
+
+
+
+
+
```
Comment (Irirwanirira) [inline src/screens/Auth/EmailLoginScreen.tsx]: Kindly remove the empty spaces present in your codebase

**Review by Irirwanirira — CHANGES_REQUESTED** [review]
Good job so far, but they some required changes before you move on to the next step, Kindly take your time and fix them accordingly.

File: src/components/PasswordInput.tsx (hunk 1 above)
Comment (KeynesBizimana) [reply src/components/PasswordInput.tsx]: Done

File: src/screens/Auth/EmailLoginScreen.tsx (hunk 3 above)
Comment (KeynesBizimana) [reply src/screens/Auth/EmailLoginScreen.tsx]: Done

File: src/components/PhoneNumberInput.tsx (hunk 2 above)
Comment (KeynesBizimana) [reply src/components/PhoneNumberInput.tsx]: Done

**Review by Irirwanirira — APPROVED** [review]
🚀 good Job, now you are good to go to the next step 🚀

**Pull request merged**